OPHD/Technology/ResearchTracker.h

```diff
@@ -37,6 +37,8 @@ class ResearchTracker
 	void updateResearch(int techId, int progress, int assigned);
 	const std::tuple<int, int>& researchProgress(int techId) const;
 
+	void clear();

```

Comment:
It might be better to re-initialize an object, rather than use a `clear()` method. If you add fields to a class, then you must update `clear()` or you may end up with subtle bugs. It's generally safer to re-initialize.
```cpp
researchTracker = ResearchTracker{}; // Construct new, and assign over top of old instance
```

From another perspective, the constructor already is a method to fully initialize a new blank object.

As a bonus, it may encourage using a non-default constructor to actually initialize the object, rather than just create a new empty object. For instance, when loading a saved game, if the list of completed and in progress techs is already parsed, you could initialize with:
```cpp
researchTracker = ResearchTracker{completedTechs, inProgressTechs};
```